src/solc_0.8/polygon/child/asset/PolygonAssetV2.sol

```diff
@@ -39,18 +39,11 @@ contract PolygonAssetV2 is ERC1155ERC721 {
         (uint256[] memory ids, uint256[] memory amounts, bytes memory data) =
             abi.decode(depositData, (uint256[], uint256[], bytes));
         address sender = _msgSender();
-        bytes32 hash = abi.decode(data, (bytes32));
+        bytes32[] memory hashes = abi.decode(data, (bytes32[]));
         for (uint256 i = 0; i < ids.length; i++) {
-            // ERC-721
-            if ((amounts[i] == 1) && (ids[i] & IS_NFT > 0)) {
-                uint8 rarity = 0;
-                _mint(hash, amounts[i], rarity, sender, user, ids[0], data, false);
-            }
-            // ERC-1155
-            else {
-                uint8 rarity = 0;
-                _mint(hash, amounts[i], rarity, sender, user, ids[0], data, false);
-            }
+            // @review - any reason to have separate calls to _mint() for ERC721 & ERC1155?
+            uint8 rarity = 0;
+            _mint(hashes[i], amounts[i], rarity, sender, user, ids[i], data, false);
```

Comment:
Not sure if this is the right place, but I'm still worried about the hashes here, we should have a test that checks that if we do a batch transfer of different asset mint batches keeps the same hash for each of them

ex:
- mintBatch 1:
  - id0: uri1/0.json
  - id1: uri1/1.json
  - id2: uri1/2.json
- mintBatch 2:
  - id3: uri2/0.json
  - id4: uri2/1.json

Then if we transfer between L1<>L2 the following batch:
- id2: uri1/2.json
- id3: uri2/0.json

The URIs should be the same as the original batch

Reply:
Yes I'll be adding more extensive tests for these scenarios